### PR TITLE
Add visual prompting zero-shot learning (`export`, IR inference)

### PR DESCRIPTION
**Pull request by sungchul1**
### Summary

This PR includes:
- [x] Enable `otx export` : export image encoder, prompt getter, and decoder to onnx and openvino IR
    - [x] Change mo CLI to API (https://github.com/openvinotoolkit/training_extensions/pull/2706#discussion_r1425459267)
- [x] openvino inferencer for IR inference
- [x] Add unit tests
- [x] Add integration tests 
- [x] Add e2e tests

TODOs
- `otx optimize`
- docstring & documentation
- Update data pipeline

### How to test

```python
# unit
$ tox -vv -e unittest-all-py310-pt2 -- tests/unit/algorithms/visual_prompting
Name                                                                                                                   Stmts   Miss  Cover   Missing
----------------------------------------------------------------------------------------------------------------------------------------------------
src/otx/algorithms/visual_prompting/adapters/openvino/model_wrappers/openvino_models.py                                   96      2    98%   55, 138
src/otx/algorithms/visual_prompting/adapters/pytorch_lightning/callbacks/inference.py                                     55      0   100%
src/otx/algorithms/visual_prompting/adapters/pytorch_lightning/config/visual_prompting_config.py                          43      2    95%   108-109
src/otx/algorithms/visual_prompting/adapters/pytorch_lightning/datasets/dataset.py                                       139     13    91%   105, 135-137, 192, 196, 231, 399-405, 459-462
src/otx/algorithms/visual_prompting/adapters/pytorch_lightning/datasets/pipelines/sam_transforms.py                       39      6    85%   36-43
src/otx/algorithms/visual_prompting/adapters/pytorch_lightning/datasets/pipelines/transforms.py                           40      1    98%   60
src/otx/algorithms/visual_prompting/adapters/pytorch_lightning/models/backbones/tiny_vit.py                              333      4    99%   223, 395, 509, 685
src/otx/algorithms/visual_prompting/adapters/pytorch_lightning/models/backbones/vit.py                                   122      0   100%
src/otx/algorithms/visual_prompting/adapters/pytorch_lightning/models/decoders/sam_mask_decoder.py                       153      1    99%   194
src/otx/algorithms/visual_prompting/adapters/pytorch_lightning/models/encoders/sam_image_encoder.py                       14      0   100%
src/otx/algorithms/visual_prompting/adapters/pytorch_lightning/models/encoders/sam_prompt_encoder.py                      94      0   100%
src/otx/algorithms/visual_prompting/adapters/pytorch_lightning/models/utils/layer_norm.py                                 14      0   100%
src/otx/algorithms/visual_prompting/adapters/pytorch_lightning/models/utils/mlp_block.py                                  10      0   100%
src/otx/algorithms/visual_prompting/adapters/pytorch_lightning/models/visual_prompters/segment_anything.py               225      4    98%   224, 576-578
src/otx/algorithms/visual_prompting/adapters/pytorch_lightning/models/visual_prompters/zero_shot_segment_anything.py     269     25    91%   196-197, 200-201, 204-205, 277, 285, 300, 303, 364, 415, 423, 455-464, 473-474, 506, 509, 548, 609, 658, 662
src/otx/algorithms/visual_prompting/configs/base/configuration.py                                                         39      0   100%
src/otx/algorithms/visual_prompting/configs/base/configuration_enums.py                                                    4      0   100%
src/otx/algorithms/visual_prompting/configs/sam_tiny_vit/configuration.py                                                  4      4     0%   7-13
src/otx/algorithms/visual_prompting/configs/sam_tiny_vit/ptq_optimization_config.py                                        5      5     0%   2-22
src/otx/algorithms/visual_prompting/configs/sam_vit_b/configuration.py                                                     4      4     0%   7-13
src/otx/algorithms/visual_prompting/configs/sam_vit_b/ptq_optimization_config.py                                           5      0   100%
src/otx/algorithms/visual_prompting/configs/zero_shot_sam_tiny_vit/configuration.py                                        4      4     0%   7-13
src/otx/algorithms/visual_prompting/configs/zero_shot_sam_tiny_vit/ptq_optimization_config.py                              5      5     0%   2-22
src/otx/algorithms/visual_prompting/tasks/inference.py                                                                   320     28    91%   102, 187, 230-231, 236, 351, 361, 366, 401, 421-422, 451, 455-464, 471-472, 476-477, 573, 583, 588, 624, 685
src/otx/algorithms/visual_prompting/tasks/openvino.py                                                                    344     26    92%   280, 312, 317-319, 321, 343, 376, 499, 503-505, 532-533, 537-538, 578, 641, 644, 664, 667, 681, 707, 716-718
src/otx/algorithms/visual_prompting/tasks/train.py                                                                        41      4    90%   90-91, 98-99
src/otx/algorithms/visual_prompting/utils/visual_prompting_utils.py                                                        3      3     0%   17-22

# integration
$ tox -vv -e tests-all-py310-pt2 -- tests/integration/cli/visual_prompting/test_visual_prompting.py
============================================= 16 passed, 4 skipped, 1 warning in 181.10s (0:03:01) =============================================

$ tox -vv -e tests-all-py310-pt2 -- tests/integration/cli/visual_prompting/test_zero_shot.py
================================================= 7 passed, 2 warnings in 101.46s (0:01:41) ==================================================

# e2e
$ tox -vv -e tests-all-py310-pt2 -- tests/e2e/cli/visual_prompting/test_visual_prompting.py
================================================== 20 passed, 1 warning in 315.07s (0:05:15) ===================================================

$ tox -vv -e tests-all-py310-pt2 -- tests/e2e/cli/visual_prompting/test_zero_shot.py
============================================ 6 passed, 3 skipped, 2 warnings in 80.43s (0:01:20) =============================================
```

### Checklist

- [x] I have added unit tests to cover my changes.​
- [x] I have added integration tests to cover my changes.​
- [x] I have added e2e tests for validation.
- [x] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/algorithms/visual_prompting/tasks/inference.py [hunk 1]
```diff
@@ -547,6 +547,151 @@ def infer(self, dataset: DatasetEntity, inference_parameters: InferenceParameter
...
+        }
+        self._export_to_onnx(onnx_path)
+
+        if export_type == ExportType.ONNX:
+            for module, path in onnx_path.items():
+                with open(path, "rb") as file:
+                    output_model.set_data(f"{module}.onnx", file.read())
+        else:
+            for module, path in onnx_path.items():
+                optimize_command = [
```
Comment (sovrasov) [inline src/otx/algorithms/visual_prompting/tasks/inference.py]: Could you try python MO API? It looks more reliable than arranging a CLI command
```
import openvino as ov
import openvino.tools.mo as mo

input_path = "/path_to_model/yolov8x.onnx"

# Convert model and perform input and output cut
input_name = "/model.2/Concat_output_0"
output_name = "/model.22/Concat_3_output_0"
ov_model = mo.convert_model(input_path, input=input_name, output=output_name)
ov.save_model(ov_model, 'model.xml')
# Compile model
ov.compile_model(ov_model)
```

**Review by MarkByun — COMMENTED** [review]
There are some hard coded constance values in functions. Could you please consider changing them as default parameter or some other configurable method for scalability?

**Comment by sungchul1** [comment]
> There are some hard coded constance values in functions. Could you please consider changing them as default parameter or some other configurable method for scalability?

Fixed at https://github.com/openvinotoolkit/training_extensions/pull/2706/commits/d2891e0ef2d52131dda894ee7025bfd7d11c81ab. Thanks for the advice.

**Review by MarkByun — DISMISSED** [review]
It looks good.

**Review by MarkByun — APPROVED** [review]

**Review by wonjuleee — APPROVED** [review]
LGTM

**Pull request merged**